overlord/state/change.go

```diff
@@ -520,6 +520,33 @@ func (c *Change) AbortLanes(lanes []int) {
 	c.abortLanes(lanes, make(map[int]bool), make(map[string]bool))
 }
 
+// AbortUnreadyLanes aborts the tasks from lanes that aren't fully ready, where
+// a ready lane is one in which all tasks a ready.
+func (c *Change) AbortUnreadyLanes() {
+	c.state.writing()
+	c.abortUnreadyLanes()
+}
+
+func (c *Change) abortUnreadyLanes() {
+	lanesWithLiveTasks := map[int]bool{}
+
+	for _, tid := range c.taskIDs {
+		t := c.state.tasks[tid]
+		ready := t.Status().Ready()
+		for _, tlane := range t.Lanes() {
+			if !ready {
```

Comment:
`!ready` could be checked before the loop, no?

Reply:
yeah, good catch, pushed an update